railties/test/application/generators_test.rb

```diff
@@ -32,7 +32,7 @@ def with_bare_config
 
     test "allow running plugin new generator inside Rails app directory" do
       rails "plugin", "new", "vendor/plugins/bukkits"
-      assert File.exist?(File.join(rails_root, "vendor/plugins/bukkits/test/dummy/config/application.rb"))
+      assert File.exist?(File.join(rails_root, "vendor/plugins/bukkits/bukkits.gemspec"))
     end
```

Comment:
Since we have no failing tests for #18073, it might be good to add some, to ensure it does not regress.